File: DesignPatternsExercises/ruby/builder_design_pattern/solution/computer_engineer.rb
require_relative 'computer_builder'

class ComputerEngineer
  # attr_reader __________(5)__________
  attr_reader :computer_builder

  def initialize
    # @computer_builder = __________(6)__________
    @computer_builder = ComputerBuilder.new
  end

  def build_computer
    # @computer_builder.__________(7)__________
    @computer_builder.computer
  end
end

File: DesignPatternsExercises/java/BuilderDesignPattern/Exercise/ComputerEngineer.java
public class ComputerEngineer {
    private ComputerBuilder computerBuilder;

    public ComputerEngineer() {
        computerBuilder = __________(6)__________();
    }

    public Computer buildComputer() {
        return computerBuilder.__________(7)__________();
    }

    public PCBuilder getComputerBuilder() {
        return computerBuilder;
    }
}
